### PR TITLE
Adds ability to launch multiple jobs on cluster with different code changes

### PR DESCRIPTION
**Pull request by pascal-roth**
# Description

Currently, for multiple runs the code is pushed to the same location on the cluster. In the case where the runs are not directly started (i.e., are in the queue), the code will be replaced each time a new job is submitted. This PR creates for each job a new copy of the code to make sure that the intended version is used for the submitted job. 

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run all the tests with `./isaaclab.sh --test` and they pass
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by hhansen-bdai — APPROVED** [review]
Looks nice! Couldn't test but seems reasonable

File: docs/source/deployment/cluster.rst [hunk 1]
```diff
@@ -76,16 +76,22 @@ The following describes the parameters that need to be configured:
...
+       with the date and time of the job submission. This allows to run multiple jobs with different code versions at
+       the same time.
    * - CLUSTER_LOGIN
      - The login to the cluster. Typically, this is the user and cluster names,
        e.g., ``your_user@euler.ethz.ch``.
    * - CLUSTER_SIF_PATH
      - The path on the cluster where the singularity image will be stored. The image will be
        copied to the compute node but not uploaded again to the cluster when a job is submitted.
+   * - REMOVE_CODE_COPY_AFTER_JOB
+     - Decide if the code copy should be removed after the job is finished. The logs from the job will not be deleted
```
Comment (Mayankm96) [inline docs/source/deployment/cluster.rst]:
```suggestion
     - Whether the copied code should be removed after the job is finished or not. The logs from the job will not be deleted
```

File: docs/source/deployment/cluster.rst [hunk 2]
```diff
@@ -76,16 +76,22 @@ The following describes the parameters that need to be configured:
        and mounted into the singularity container. This should increase the speed of starting
        the simulation.
    * - CLUSTER_ISAACLAB_DIR
-     - The directory on the cluster where the Isaac Lab code is stored. This directory has to
+     - The directory on the cluster where the Isaac Lab logs are stored. This directory has to
        end on ``isaaclab``. It will be copied to the compute node and mounted into
        the singularity container. When a job is submitted, the latest local changes will
-       be copied to the cluster.
+       be copied to the cluster to a new directory in the format ``${CLUSTER_ISAACLAB_DIR}_${date}_${time}``
```
Comment (Mayankm96) [inline docs/source/deployment/cluster.rst]:
```suggestion
       be copied to the cluster to a new directory in the format ``${CLUSTER_ISAACLAB_DIR}_${datetime}``
```

**Review by Mayankm96 — APPROVED** [review]
No way for me to test this as well but if it works for you then sure.

Also need to modify the PBS script?

**Comment by pascal-roth** [comment]
@Mayankm96 adjusted the PBS file, PR should be ready to merge now

**Pull request merged**